llm-ollama.el

```diff
@@ -116,6 +118,37 @@ STREAMING if non-nil, turn on response streaming."
     (push `("prompt" . ,(string-trim (llm-chat-prompt-interaction-content
                                       (car (last (llm-chat-prompt-interactions prompt))))))
           request-alist)
+    request-alist))
+
+(defun llm-ollama--chat-request-chat (prompt)
+  "Create the chat request data for the `chat' endpoint.
+PROMPT is the chat prompt to process.
+Returns REQUEST-ALIST with prompt request data."
+  (let (request-alist messages)
+    (setq messages
+          (mapcar (lambda (interaction)
+                    `(("role" . ,(symbol-name (llm-chat-prompt-interaction-role interaction)))
+                      ("content" . ,(llm-chat-prompt-interaction-content interaction))))
+                  (llm-chat-prompt-interactions prompt)))
+    (when (llm-chat-prompt-context prompt)
+      (push `(("role" . "system")
+              ("content" . ,(llm-provider-utils-get-system-prompt prompt llm-ollama-example-prelude)))
+            messages))
+    (push `("messages" . ,messages) request-alist)
+    request-alist))
+
+(defun llm-ollama--chat-request (provider prompt)
+  "From PROMPT, create the chat request data to send.
+PROVIDER is the llm-ollama provider to use.
+
+Calls the appropriate helper function to initialize
+`request-alist' depending on whether a `chat' or `generate'
+endpoint is specified in the provider."
+  (let* ((request-alist 
+          (if (string= (llm-ollama-endpoint provider) "chat")
```

Comment:
You can just use "equal" here, string= is more about comparing without properties.